custom_components/tahoma/cover.py

```diff
@@ -239,104 +211,62 @@ def icon(self):
-
-        if COMMAND_UP in self.tahoma_device.command_definitions:
-            return self.apply_action(COMMAND_UP)
+        self.apply_action(self.select_command([COMMAND_OPEN, COMMAND_UP]))
 
     def open_cover_tilt(self, **kwargs):
         """Open the cover tilt."""
-
-        if COMMAND_OPEN_SLATS in self.tahoma_device.command_definitions:
-            return self.apply_action(COMMAND_OPEN_SLATS)
+        self.apply_action(self.select_command([COMMAND_OPEN_SLATS]))
 
     def close_cover(self, **kwargs):
         """Close the cover."""
-
-        if COMMAND_CLOSE in self.tahoma_device.command_definitions:
-            return self.apply_action(COMMAND_CLOSE)
-
-        if COMMAND_DOWN in self.tahoma_device.command_definitions:
-            return self.apply_action(COMMAND_DOWN)
+        self.apply_action(self.select_command([COMMAND_CLOSE, COMMAND_DOWN]))
 
     def close_cover_tilt(self, **kwargs):
         """Close the cover tilt."""
-
-        if COMMAND_CLOSE_SLATS in self.tahoma_device.command_definitions:
-            return self.apply_action(COMMAND_CLOSE_SLATS)
+        self.apply_action(self.select_command([COMMAND_CLOSE_SLATS]))
 
     def stop_cover(self, **kwargs):
         """Stop the cover."""
-
-        if COMMAND_STOP in self.tahoma_device.command_definitions:
-            return self.apply_action(COMMAND_STOP)
-
-        if COMMAND_STOP_IDENTIFY in self.tahoma_device.command_definitions:
-            return self.apply_action(COMMAND_STOP_IDENTIFY)
-
-        if COMMAND_MY in self.tahoma_device.command_definitions:
-            return self.apply_action(COMMAND_MY)
+        self.apply_action(
+            self.select_command([COMMAND_STOP, COMMAND_STOP_IDENTIFY, COMMAND_MY])
+        )
 
     def stop_cover_tilt(self, **kwargs):
         """Stop the cover."""
-
-        if COMMAND_STOP_IDENTIFY in self.tahoma_device.command_definitions:
-            return self.apply_action(COMMAND_STOP_IDENTIFY)
-
-        if COMMAND_STOP in self.tahoma_device.command_definitions:
-            return self.apply_action(COMMAND_STOP)
-
-        if COMMAND_MY in self.tahoma_device.command_definitions:
-            return self.apply_action(COMMAND_MY)
+        self.apply_action(
+            self.select_command([COMMAND_STOP_IDENTIFY, COMMAND_STOP, COMMAND_MY])
+        )
 
     @property
     def supported_features(self):
         """Flag supported features."""
 
         supported_features = 0
 
-        if COMMAND_OPEN_SLATS in self.tahoma_device.command_definitions:
+        if self.select_command([COMMAND_OPEN_SLATS]):
             supported_features |= SUPPORT_OPEN_TILT
 
-            if (
-                COMMAND_STOP in self.tahoma_device.command_definitions
-                or COMMAND_STOP_IDENTIFY in self.tahoma_device.command_definitions
-                or COMMAND_MY in self.tahoma_device.command_definitions
-            ):
+            if self.select_command([COMMAND_STOP_IDENTIFY, COMMAND_STOP, COMMAND_MY]):
```

Comment:
Here you you biase the usage of your function. I prefer something more straightforward:
```python
if {COMMAND_STOP, COMMAND_STOP_IDENTIFY, COMMAND_MY} & set(self.tahoma_device.command_definitions):
  ...
```

Reply:
isn't that good enough? 
```
if self.has_command(COMMAND_STOP_IDENTIFY, COMMAND_STOP, COMMAND_MY):
```

Reply:
Yes it's good for me.